client/battle/BattleInterface.cpp

```diff
@@ -336,11 +336,16 @@ void BattleInterface::battleFinished(const BattleResult& br, QueryID queryID)
 
 void BattleInterface::spellCast(const BattleSpellCast * sc)
 {
-	windowObject->blockUI(true);
+	// Do not deactivate anything in tactics mode
+	// This is battlefield setup spells
+	if(!tacticsMode)

```

Comment:
Why is this needed? UI should be reactivated once spellcast ends

Reply:
@IvanSavenko this is not the case if there is no spellcaster hero and no spellcaster unit.

Reply:
In such case no EndAction or any other spell action called, so, interface is just locked and not unlocked next (because in tactics mode there is no speed system to switch to next creature after action)/

Reply:
Why?
Server should still send notification that spell ended.

Does moat casting bypasses BattleSpellMechanics::cast?

Reply:
No, it uses BattleSpellMechanics::cast precisely. But it does not use BattleAction, like opening battle spells. But because opening battle spells runs after tactics, this are not suspectible to this bug.

Reply:
```BattleSpellMechanics::cast``` use ```BattleSpellCast``` package, so, we react in client to it inside ```BattleInterface::spellCast``, in which we try to block interface twice (in blockUI and deactivateStack), and will not unlock it in any way inside this function. Unlocking usually happens in another function, which will not run in tactics mode.

Reply:
> we should can blow up Tower mines by usage of tactics

I'd call that a bug...

Reply:
@IvanSavenko this is controversal. If we move a creature in tactics mode, and its pathfinder will create a path through minefield, then mines should be blown up (according to FizMig). I do not see it as a bug, because a setup of moats and mines should be done before battle, and not inside it.

Reply:
I say VCMI should not implement every H3 bug.
If a bug seems to be crucial and needs to be implemented, should be discussed before.

Reply:
Tactics mode supports instant teleportig of creatures. Which invalidates this implementation.

Reply:
Interesting, if there were a bug that allows catapult to attack walls during tactics that would be controversial as well?

After all, setup of walls and towers should be done before battle, and no during it?

Reply:
But setup of tower shooters is done before battle, why moat setup should be done after?

Reply:
I would rather say that 
a) pathfinder should avoid moat hexes in tactics
b) moat effects should not trigger during tactics

Defender needs to see tower mines if he has tactics

Reply:
If defender need to see tower mines - then it should be cast before tactics.